### PR TITLE
[CHE-155] Create PostgreSQL Dev Environment

### PR DESCRIPTION
**Pull request by brok3turtl3**
### Description

The PR was originally intended to set up the docker dev environment postgres container and ensure connectivity to the app as part of the [CHE-140](https://codehammers.atlassian.net/browse/CHE-140?atlOrigin=eyJpIjoiNDZkNDZmYTYxNjkzNGZlZGI3YTEzODg0ODc3MWJiN2YiLCJwIjoiaiJ9) epic to create our job applications tracker. I went 'slightly' out of scope on this and also:

1. Created a script to seed database with sample applications
2. Created Router and a basic endpoint to fetch all applications
3. Created an Applications page for front end and hooked up Router navigation
4. Created a basic fetch and display from front end to confirm full stack connection

### Jira Task

[CHE-155](https://codehammers.atlassian.net/browse/CHE-155?atlOrigin=eyJpIjoiOThkNTljYzQ1MTdkNGZiYjllOGFhODZiMDRjNGU1ZDAiLCJwIjoiaiJ9)

### Testing Instructions

1. Ensure branches are all up to date
2. Use npm run docker-remove-all nd then confirm in Docker desktop all container, images and volumes have been removed.
3. Use nom run docker-dev from this feature branch
4. Log into application and make sure that there are no errors and that the Applications page loads and displays the seed data as expected
5. ***You will need to execute Step 2 and 3 again when you return to your dev branch to make sure you are using the correct images for your branch***

### Checklist

- [x] I added a descriptive title to this PR.
- [x] I filled out the **Description**, **Jira Task**, and **Testing Instructions** sections above.
- [ ] I added or updated [Jest unit tests]for any changes to components, server-side controllers, etc.
- [x] I ran `npm run docker-test` in my local environment to check that this PR passes all unit tests.
- [x] I did a quick check to make sure my code changes follow the recommended style guide.

### Additional Notes, Images, etc.

Screenshot 2024-05-16 at 9 18 27 AM

[CHE-140]: https://codehammers.atlassian.net/browse/CHE-140?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CHE-155]: https://codehammers.atlassian.net/browse/CHE-155?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Dockerfile-postgres [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM postgres:16.3
+
+COPY ./scripts/sql_db_init.sql /docker-entrypoint-initdb.d/
+
+
+
```
Comment (jimpulsee) [inline Dockerfile-postgres]: Why the trailing lines? Is there a lint configuration we can set to catch this?

File: client/src/AuthenticatedApp.tsx [hunk 2]
```diff
@@ -8,6 +8,7 @@ import Profile from "./pages/Profile/Profile";
 import EditProfilePage from "./pages/EditProfilePage/EditProfilePage";
 import Directory from "./pages/DirectoryPage/DirectoryPage";
 import NotFoundPage from "./pages/NotFoundPage/NotFoundPage";
+import ApplicationsPage from "./pages/ApplicationsPage/ApplicationsPage";
 import { useNavigate } from "react-router-dom";
```
Comment (jimpulsee) [inline client/src/AuthenticatedApp.tsx]: Consolidate this import. Is there a lint configuration we can set to catch this?

File: client/src/components/Header/Header.tsx [hunk 3]
```diff
@@ -54,6 +54,14 @@ const Header = (): JSX.Element => {
...
             Alumni
           </Link>
+          <Link
+            to="/app/applications"
+            className={`text-lg md:text-xl ${
+              currentPath === "main" ? "text-gray-300" : "hover:text-gray-300"
+            } transition transform hover:scale-105`}
+          >
+            Applications
+          </Link>
```
Comment (jimpulsee) [inline client/src/components/Header/Header.tsx:64]: A small task can be to refactor `<Link />` components to `<NavLink />`. It is built to handle logic depending on current path.

File: client/src/pages/ApplicationsPage/ApplicationsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    async function fetchApplications() {
+      try {
+        const response = await axios.get("/api/applications");
+        setApplications(response.data);
+      } catch (error) {
+        console.error("Error fetching applications:", error);
+      }
+    }
+
+    fetchApplications();
```
Comment (jimpulsee) [inline client/src/pages/ApplicationsPage/ApplicationsPage.tsx:20]: Thoughts on using [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) for async functions defined in useEffect?

File: docker-compose-dev.yml
Comment (jimpulsee) [inline docker-compose-dev.yml:1]: Do we need to keep comments?

File: server/controllers/applicationsController.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { Request, Response, NextFunction } from "express";
+import { pool } from "../config/sql-db";
+
+const getAllApplications = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    console.log("Middleware hit!!!");
```
Comment (jimpulsee) [inline server/controllers/applicationsController.ts]: Remove

**Review by jimpulsee — CHANGES_REQUESTED** [review]

File: client/src/AuthenticatedApp.tsx (hunk 2 above)
Comment (brok3turtl3) [reply client/src/AuthenticatedApp.tsx]: Great catch - done!

File: Dockerfile-postgres (hunk 1 above)
Comment (brok3turtl3) [reply Dockerfile-postgres]: Someone is working on it now =)

File: client/src/pages/ApplicationsPage/ApplicationsPage.tsx (hunk 4 above)
Comment (brok3turtl3) [reply client/src/pages/ApplicationsPage/ApplicationsPage.tsx:20]: @JimmyTran48 So my understanding is that the biggest benefit of using IIFE's is to help with scoping and polluting. In this case it's all contained within the useEffect. I don't *think* there would be any significant increase in performance switching it. Am I missing something?

File: client/src/components/Header/Header.tsx (hunk 3 above)
Comment (brok3turtl3) [reply client/src/components/Header/Header.tsx:64]: Great suggestion @JimmyTran48. Adding a ticket!

**Review by jimpulsee — APPROVED** [review]

**Pull request merged**